Rdmp.Core.Tests/DataLoad/Engine/Integration/RemoteDatabaseAttacherTests.cs

```diff
@@ -102,6 +104,178 @@ public void TestRemoteDatabaseAttach(DatabaseType dbType, Scenario scenario)
+    [TestCase(DatabaseType.MicrosoftSQLServer, Scenario.AllColumns, AttacherHistoricalDurations.DeltaReading)]
+    [TestCase(DatabaseType.MicrosoftSQLServer, Scenario.MissingPreLoadDiscardedColumn, AttacherHistoricalDurations.DeltaReading)]
+    [TestCase(DatabaseType.MicrosoftSQLServer, Scenario.MissingPreLoadDiscardedColumnButSelectStar, AttacherHistoricalDurations.DeltaReading)]
+    public void TestRemoteDatabaseAttacherWithDateFilter(DatabaseType dbType, Scenario scenario, AttacherHistoricalDurations duration)
+    {
+        var db = GetCleanedServer(dbType);
+
+        var dt = new DataTable();
+        dt.Columns.Add("animal");
+        dt.Columns.Add("date_seen");
+        var withinDate = Within(duration);
+        dt.Rows.Add("Cow", withinDate);
+        dt.Rows.Add("Crow", Outwith(duration));
+
+        var tbl = db.CreateTable("MyTable", dt);
+
+        Assert.That(tbl.GetRowCount(), Is.EqualTo(2));
+        Import(tbl, out var ti, out _);
+        //Create a virtual RAW column
+        if (scenario is Scenario.MissingPreLoadDiscardedColumn or Scenario.MissingPreLoadDiscardedColumnButSelectStar)
+            new PreLoadDiscardedColumn(CatalogueRepository, ti, "MyMissingCol");
+
+        var externalServer = new ExternalDatabaseServer(CatalogueRepository, "MyFictionalRemote", null);
+        externalServer.SetProperties(db);
+
+        var attacher = new RemoteDatabaseAttacher();
+        attacher.Initialize(null, db);
+        attacher.HistoricalFetchDuration = duration;
+        attacher.RemoteTableDateColumn = "date_seen";
+        attacher.LoadRawColumnsOnly = scenario is Scenario.AllRawColumns or Scenario.MissingPreLoadDiscardedColumn;
+        attacher.RemoteSource = externalServer;
+
+        if (duration == AttacherHistoricalDurations.Custom)
+        {
+            attacher.CustomFetchDurationStartDate = DateTime.Now.AddDays(-7);
+            attacher.CustomFetchDurationEndDate = DateTime.Now;
+        }
+
+        if (duration == AttacherHistoricalDurations.DeltaReading)
+        {
+            attacher.DeltaReadingStartDate = DateTime.Now.AddDays(-7);
+            attacher.DeltaReadingLookBackDays = 0;
+            attacher.DeltaReadingLookForwardDays = 5;
+        }
+
+
+        var lm = new LogManager(CatalogueRepository.GetDefaultFor(PermissableDefaults.LiveLoggingServer_ID));
+        lm.CreateNewLoggingTaskIfNotExists("amagad");
+        var dli = lm.CreateDataLoadInfo("amagad", "p", "a", "", true);
+
+        var job = NSubstitute.Substitute.For<IDataLoadJob>();
+        job.RegularTablesToLoad.Returns(new List<ITableInfo> { ti });
+        job.LookupTablesToLoad.Returns(new List<ITableInfo>());
+        job.DataLoadInfo.Returns(dli);
+
+        if (duration == AttacherHistoricalDurations.SinceLastUse)
+        {
+            job.LoadMetadata.LastLoadTime = DateTime.Now.AddDays(-1);// last used yesterday
+            job.LoadMetadata.SaveToDatabase();
+        }
+
+        switch (scenario)
```

Comment:
Planning to add more here later? Otherwise a simple "if scenario==Scenario.MissingPreLoadDiscardedColumn" would make more sense than a switch with all empty branches?